### PR TITLE
Create Product entity

### PR DESCRIPTION
**Pull request by botirjonabdumannonovv**
Close: #8

### CODE REVIEW

**Review by rashidovasadbek — APPROVED** [review]

**Review by isxoqovxasanboy — DISMISSED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Product.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using Feedback.Analyzer.Domain.Common.Entities;
+
+namespace Feedback.Analyzer.Domain.Entities;
+
```
Comment (isxoqovxasanboy) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Product.cs:4]: add summary

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Product.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/// <summary>
+/// Represents a product entity, inheriting auditable properties.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Product.cs:6]: Remove "inheriting auditable properties."

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/ProductConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Feedback.Analyzer.Persistence.EntityConfiguration;
+
+public class ProductConfiguration : IEntityTypeConfiguration<Product>
+{
+    public void Configure(EntityTypeBuilder<Product> builder)
+    {
+        builder.Property(organization => organization.Name).IsRequired().HasMaxLength(128);
+        builder.Property(organization => organization.Description).IsRequired().HasMaxLength(512);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/EntityConfiguration/ProductConfiguration.cs:12]: Change lambda variable name to "product"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IProductRepository.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
+using System.Linq.Expressions;
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+/// <summary>
+/// Represents a repository for managing products.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IProductRepository.cs:9]: Change to "Defines"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/ProductRepository.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.DataContexts;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+namespace Feedback.Analyzer.Persistence.Repositories;
+
+/// <summary>
+/// Represents a repository for managing product data.
+/// </summary>
+/// <param name="dbContext"></param>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/ProductRepository.cs:13]: Remove params from primary constructors

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by isxoqovxasanboy — DISMISSED** [review]

**Review by sulton-max — APPROVED** [review]
Good job

**Review by bunyodkamalov — APPROVED** [review]

**Review by isxoqovxasanboy — APPROVED** [review]

**Pull request merged**